src/txt.rs

```diff
@@ -21,55 +22,104 @@ impl TXTRecord {
         }
     }
 
-    /// Sets a key/value pair with given strings
-    pub fn set_value(&mut self, key: &str, value: &str) -> Result<(), DNSServiceError> {
-        self.set_value_bytes(key, value.as_bytes())
+    /// Sets a key/value pair
+    /// 
+    /// **Note:** Only the first 256 bytes of the value will be used.
+    pub fn set_value<V>(&mut self, key: &str, value: Option<V>) -> Result<(), DNSServiceError>
+        where V: AsRef<[u8]>
+    {
+        let value = value.as_ref().map(|x| x.as_ref());
+        let key = CString::new(key).or(Err(DNSServiceError::InvalidString))?;
+        let value_size = value.map_or(0, |x| x.len().min(u8::max_value() as usize) as u8);
+        let result = unsafe {
+            TXTRecordSetValue(&mut self.raw, key.as_ptr(), value_size, value.map_or(ptr::null(), |x| x.as_ptr() as *const c_void))
+        };
+        if result == kDNSServiceErr_NoError {
+            return Ok(());
+        }
+        Err(DNSServiceError::ServiceError(result))
     }
 
-    /// Sets a key/value pair to a raw bytes value
-    pub fn set_value_bytes(&mut self, key: &str, value: &[u8]) -> Result<(), DNSServiceError> {
+    /// Removes a key/value pair
+    pub fn remove(&mut self, key: &str) {
+        let key = match CString::new(key) {
+            Ok(key) => key,
+            Err(_) => {
+                // If the key contains an interior NUL then we know we don't contain it, and
+                // therefore there's nothing to remove
+                return;
+            }
+        };
+        // NB: The only error that TXTRecordRemoveValue can return is one signifying that the key
+        // did not exist.
         unsafe {
-            let key = CString::new(key).map_err(|_| DNSServiceError::InvalidString)?;
-            let value_size = value.len() as u8;
-            let result = TXTRecordSetValue(&mut self.raw, key.as_ptr(), value_size, value.as_ptr() as *mut c_void);
-            if result == kDNSServiceErr_NoError {
-                return Ok(());
+            TXTRecordRemoveValue(&mut self.raw, key.as_ptr());
+        }
+    }
+
+    /// Checks if a key exists
+    pub fn contains_key(&mut self, key: &str) -> bool {
```

Comment:
Does this still have to be &must self to work? Since ideally we’re just reading

Reply:
No, I just missed that one. This can be immutable as well. I'll update the PR.